libs/blocks/preflight/checks/asoApi.js

```diff
@@ -0,0 +1,275 @@
+import {
+  SEO_TITLES,
+  STATUS,
+  SEO_IDS,
+  SEO_DESCRIPTIONS,
+  ASO_POLL_INTERVAL_MS,
+  ASO_MAX_RETRIES,
+  ASO_TIMEOUT_MS,
+} from './constants.js';
+
+const CHECK_API = 'https://spacecat.experiencecloud.live/api/v1';
+
+export const asoCache = {
+  identify: null,
+  identifyFinished: false,
+  suggest: null,
+  suggestFinished: false,
+  sessionToken: null,
+};
+
+const lanaLog = (message) => {
+  window.lana.log(message, {
+    sampleRate: 1,
+    tags: 'preflight',
+    errorType: 'e',
+  });
+};
+
+async function getASOToken() {
+  window.adobeImsFactory.createIMSLib({
+    client_id: 'milo-tools',
+    scope: 'AdobeID,openid,gnav,read_organizations,additional_info.projectedProductContext,additional_info.roles',
+    environment: 'prod',
+    autoValidateToken: true,
+    useLocalStorage: false,
```

Comment:
let's use `modalMode: true,` to allow users to login without leaving the window.